enterprise/server/cmd/server/main.go

```diff
@@ -181,6 +183,21 @@ func main() {
 		realEnv.SetCacheRedisClient(redisClient)
 	}
 
+	if redisTarget := configurator.GetUsageRedisTarget(); redisTarget != "" {
+		redisClient := redisutil.NewClientWithOpts(&redis.Options{
+			Addr: redisTarget,
+			// Use db1 for usage data since we use SCAN to iterate over all usage
+			// keys and scanning over unrelated keys would hurt performance.
+			DB: 1,
```

Comment:
I don't love introducing yet another redis client (YARC?), and also the notion of scanning the entire key space.

How about using the normal redis client, and keeping track of your "usage counters" by adding them to a set every time you increment them with SADD.

So, at the end of the minute (HHMM) your keyspace would look like this:

HHMM: set(GR123-HHMM, GR345-HHMM, GR567-HHMM)
GR123-HHMM: {cas_hits: 1, ac_hits: 2}
GR345-HHMM: {cas_hits: 3, ac_hits: 4}
GR567-HHMM: {cas_hits: 5, ac_hits: 6}
... other keys you don't care about...


Your set flow would be:

Pipeline:
  SADD(HHMM, GR123-HHMM)
  INCR(GR123-HHMM-cas_hits, 1)
  EXPIRE(GR123-HHMM-cas_hits, 120)

And your read flow would be:

Get global lock()
for period in HHMM, HHMM-1, HHMM-2 {
  for counter in period:
    flushToMySQL(counter)
  DEL period